check-log/lib/check-log.go

```diff
@@ -346,20 +347,50 @@ func (opts *logOpts) match(line string) (bool, []string) {
 	return true, matches
 }
 
+type state struct {
+	SkipBytes int64 `json:"skip_bytes"`
+}
+
+func loadState(fname string) (*state, error) {
+	state := &state{}
+	_, err := os.Stat(fname)
+	if err == nil { // posfile exists
+		b, err := ioutil.ReadFile(fname)
+		if err == nil {
+			err = json.Unmarshal(b, state)
+		}
+		return state, err
+	}
+	return state, nil
+}
+
 var stateRe = regexp.MustCompile(`^([a-zA-Z]):[/\\]`)
 
 func getStateFile(stateDir, f string, args []string) string {
 	return filepath.Join(
 		stateDir,
 		fmt.Sprintf(
-			"%s-%x",
+			"%s-%x.json",
 			stateRe.ReplaceAllString(f, `$1`+string(filepath.Separator)),
 			md5.Sum([]byte(strings.Join(args, " "))),
 		),
 	)
 }
 
 func getBytesToSkip(f string) (int64, error) {
+	if _, err := os.Stat(f); err == nil {
```

Comment:
loadState itself also checks file's existence, so when json stat file exists `os.Stat` will be executed twice.
Do you have any idea to solve this?
(an idea: Make loadState return `(nil, nil)` when file does not exist, and handle nil here)

Reply:
Thanks. 
I solved at https://github.com/mackerelio/go-check-plugins/pull/252/commits/1724b13f4a5280a649f3574179ba910674ab183f.